tests/tests.c

```diff
@@ -1023,7 +1037,14 @@ bool sirtest_levelssanity(void) {
     printf(INDENT_ITEM WHITE("greater than SIRL_ALL: %04"PRIx16) "\n", invalid);
 
     /* individual invalid level. */
+#if defined(__clang__) || defined(__clang_version__)
+# pragma clang diagnostic push
+# pragma clang diagnostic ignored "-Wassign-enum"
+#endif
     sir_level invalid2 = 0x1337;
```

Comment:
Same thing with these. Does a cast squash the warning? I really, really hate these pragma blocks.

Reply:
Casting resolves that warning on AppleClang.

Reply:
Yeah, that'll work